apps/studio/src/features/editing-experience/components/PageCreateModal.tsx

```diff
@@ -67,11 +68,18 @@ export const PageCreateModal = ({
     setPageUrlLen(watchAllFields.url?.length || 0)
   }, [watchAllFields])
 
-  /* TODO: When integrating with BE */
-  const onSubmit: SubmitHandler<PageCreateFormFields> = (data) =>
-    console.log(data)
+  const [submittedData, setSubmittedData] = useState<PageCreateFormFields>({
+    title: '',
+    url: '',
+  })
+
+  // This might need to go into the layout component? or we keep state in this component then pass selectedlayout and setselectedlayout into the component
+  // Prefer to keep page creation states within the layout selection modal? More portable in that case if we want to shift the entry point.
+  const onSubmit: SubmitHandler<PageCreateFormFields> = (data) => {
+    setSubmittedData(data)
+  }
 
-  return (
+  let content = (
```

Comment:
slight antipattern, just return directly based on state instead of assigning to a variable.

```tsx
if (submittedData.title) {
  return <xxx>
}

return <yyy>
```